packages/headless/src/modules/product/services/index.ts

```diff
@@ -148,4 +152,23 @@ export class ProductService {
     if (!products) return this.helper.serviceResponse.errorResponse(GetProductsByConditionErrorMessages.CAN_NOT_GET_PRODUCTS, null, HttpStatus.BAD_REQUEST);
     return this.helper.serviceResponse.successResponse(products, HttpStatus.OK);
   }
+
+  async GetCustomizedProducts(query: GetCustomizedProductsQuery): Promise<GetCustomizedProductsResponse> {
+    const { skip, limit, tags } = query;
+    const products = await this.getProductsByTags({ tags, skip, limit });
+    if (!products) {
+      return this.helper.serviceResponse.errorResponse(GetCustomizedProductsErrorMessages.CAN_NOT_GET_CUSTOMIZED_PRODUCTS, null, HttpStatus.BAD_REQUEST);
+    }
+    return this.helper.serviceResponse.successResponse(products, HttpStatus.OK);
+  }
+
+  async getProductsByTags(condition) {
+    const { skip, limit, tags } = condition;
+    switch (tags) {
+      case GetCustomizedProductsTagsEnum.TOP_SELLING_PRODUCTS:
+        return await this.productRepo.GetTopSellingProducts({}, skip, limit);
+      default:
+        return await this.productRepo.findAllProducts({}, skip, limit);
```

Comment:
default return empty array response. you can add other tags in 'case' blocks and use same function for now.